modules/core/src/mathfuncs.cpp

```diff
@@ -1600,50 +1608,311 @@ static bool ocl_patchNaNs( InputOutputArray _a, float value )
+        }
+    }
+    else if (_a.depth() == CV_64F)
     {
-        int* tptr = ptrs[0];
-        size_t j = 0;
+        Cv64suf val;
+        val.f = _val;
 
 #if CV_SIMD
-        size_t cWidth = (size_t)v_int32::nlanes;
-        for ( ; j + cWidth <= len; j += cWidth)
+        v_int64 v_mnt_mask = vx_setall_s64(0x000FFFFFFFFFFFFF), v_exp_mask = vx_setall_s64(0x7FF0000000000000);
+        v_int64 v_val = vx_setall_s64(val.i);
+#if !CV_SIMD128_64F
+        v_int64 mask10 = vx_setall_s64(0xffffffff00000000);
+#endif
+#endif
+
+        for (size_t i = 0; i < it.nplanes; i++, ++it)
         {
-            v_int32 v_src = vx_load(tptr + j);
-            v_int32 v_cmp_mask = v_mask2 < (v_src & v_mask1);
-            v_int32 v_dst = v_select(v_cmp_mask, v_val, v_src);
-            v_store(tptr + j, v_dst);
+            int64* tptr = (int64*)ptrs[0];
+            size_t j = 0;
+
+#if CV_SIMD
+            size_t cWidth = (size_t)v_int64::nlanes;
+            for (; j + cWidth <= len; j += cWidth)
+            {
+                v_int64 v_src = vx_load(tptr + j);
+                v_int64 vande = v_src & v_exp_mask;
+                v_int64 vandm = v_src & v_mnt_mask;
+                v_int64 ve, vm;
+
+                #if CV_SIMD128_64F
+                ve = vande == v_exp_mask;
+                vm = vandm != vx_setzero_s64();
+                #else
+                // emulating 64-bit integer eq comparison: ve = (v_src & v_exp_mask) == v_exp_mask
+                {
+                    v_int32 vue32 = v_reinterpret_as_s32(vande);
+                    v_int32 vme32 = v_reinterpret_as_s32(v_exp_mask);
+                    v_int32 veq32 = vue32 == vme32;
+                    v_int32 sh1 = v_rotate_left<1>(veq32);
+                    v_int32 vand = veq32 & sh1;
+                    v_int32 sh2 = v_rotate_right<1>(vand);
+                    ve = (v_reinterpret_as_s64(vand) & mask10) | (v_reinterpret_as_s64(sh2) & ~mask10);
+                }
+                // emulating 64-bit integer ineq comparison: vm = (v_src & v_mnt_mask) != zero
+                {
+                    v_int32 vue32 = v_reinterpret_as_s32(vandm);
+                    v_int32 vme32 = vx_setzero_s32();
+                    v_int32 veq32 = vue32 != vme32;
+                    v_int32 sh1 = v_rotate_left<1>(veq32);
+                    v_int32 vor = veq32 | sh1;
+                    v_int32 sh2 = v_rotate_right<1>(vor);
+                    vm = (v_reinterpret_as_s64(vor) & mask10) | (v_reinterpret_as_s64(sh2) & ~mask10);
+                }
+                #endif
+
+                v_int64 v_isnan = ve & vm;
+                v_int64 v_dst = (v_isnan & v_val) | ((~v_isnan) & v_src);
+                v_store(tptr + j, v_dst);
+            }
+            vx_cleanup();
+#endif
+
+            for (; j < len; j++)
+                if ((tptr[j] & 0x7FFFFFFFFFFFFFFF) > 0x7FF0000000000000)
+                    tptr[j] = val.i;
         }
-        vx_cleanup();
+    }
+}
+
+#if CV_SIMD
```

Comment:
SIMD optimizations in `core` module should go to `.simd.hpp`.

Reply:
Should I add it through HAL and all SIMD dispatching mechanisms as other functions in `mathfuncs_core.simd.hpp` are done or there are easier ways?

Reply:
Other functions should be handled separately (do not touch them in this PR).

At first we need to collect performance for different ISA optimizations for added code: https://github.com/opencv/opencv/wiki/CPU-optimizations-build-options#optimization-developer-guide